system-int-test/test/cmr/system_int_test/search/granule_spatial_search_test.clj

```diff
@@ -49,6 +51,86 @@
   [& ords]
   (codec/url-encode (umm-s/set-coordinate-system :geodetic (apply polygon ords))))
 
+;; Tests all granule search when allow-all-granule-params-flag is false
+;; Existing tests should cover the case when the flag is set to true. 
+(deftest allow-all-granule-params-flag-false-search-test 
+  (let [saved-flag-value (concepts-search/allow-all-granule-params-flag)
+        saved-header-value (concepts-search/allow-all-gran-header) 
+        _ (side/eval-form `(concepts-search/set-allow-all-granule-params-flag! false))
+        _ (side/eval-form `(concepts-search/set-allow-all-gran-header! "allow-all-gran"))
+        header1 {"client-id" "testing", "allow-all-gran" true} 
+        header2 {"client-id" "testing", "allow-all-gran" false} 
+        header3 {"client-id" "testing"}
+        header4 {"allow-all-gran" true} 
+        header5 {"allow-all-gran" false}
+        header6 {}
+        params1 {:provider "PROV1"}
+        params2 {:provider_id "PROV1"}
+        params3 {:concept_id "C1234-PROV1"}
```

Comment:
This should be a granule concept-id so start with "G" instead of "C"

Reply:
I added a test for concept-id being the "G" case since concept-id could be both granule concept id and collection concept id here.